src/devtools/client/debugger/src/components/TestInfo/TestStepItem.tsx

```diff
@@ -125,50 +119,66 @@ export function TestStepItem({ step, argString, index, id }: TestStepItemProps)
           const callerFrame = frames[1];
 
           if (messageEnd?.commandVariable) {
-            const cmdResult = await client.evaluateExpression(
+            const cmdResult = await evaluateAsync(
+              client,
               endPauseResult.pauseId,
-              `${messageEnd.commandVariable}.get("subject")`,
-              callerFrame.frameId
+              callerFrame.frameId,
+              `${messageEnd.commandVariable}.get("subject")`
             );
 
-            const cmdObject = cmdResult.data.objects?.find(
-              o => o.objectId === cmdResult.returned?.object
-            );
-            const length: number | undefined = cmdObject?.preview?.properties?.find(
-              o => o.name === "length"
-            )?.value;
-            const subjects = Array.from({ length: length || 0 }, (_, i) =>
-              cmdResult.data.objects?.find(
-                obj =>
-                  obj.objectId ===
-                  cmdObject?.preview?.properties?.find(p => p.name === String(i))?.object
-              )
-            );
+            const cmdObjectId = cmdResult.returned?.object;
+
+            if (cmdObjectId) {
+              const cmdObject = await getObjectWithPreviewHelper(
+                client,
+                endPauseResult.pauseId,
+                cmdObjectId,
+                true
+              );
 
-            const nodeIds = subjects.filter(s => s?.preview?.node).map(s => s?.objectId!);
-            setSubjectNodePauseData({ pauseId: endPauseResult.pauseId, nodeIds });
+              const length: number | undefined = cmdObject?.preview?.properties?.find(
```

Comment:
:hand: Have to say this is pretty hard to read :)

If it at all helps, there's a `getJson` util I wrote that will recursively fetch contents of an object or array and return the final result as normal data.  It might be easier to use that to fetch `cmdObject` and then pull out values, if you want to try it

Reply:
Thanks for the pointer! I tried it but it was returning empty objects instead of the DOM node objects. I don't actually need the objects themselves here and and ids are sufficient but I take your point that the code was messy so I refactored it to be more straightforward.